be/src/exec/es/es_predicate.h

```diff
@@ -134,7 +134,7 @@ struct ExtFunction : public ExtPredicate {
 
 class EsPredicate {
 public:
-    EsPredicate(ExprContext* context, const TupleDescriptor* tuple_desc, ObjectPool* pool);
+    EsPredicate(ExprContext* context, const TupleDescriptor* tuple_desc, RuntimeState* runtime_state, ObjectPool* pool);
```

Comment:
RuntimeState is a relatively high-level struct. In my opinion, it is not a good idea to let EsPredicate reference RuntimeState.
Suppose you want to write a UN for EsPredicate, but you must construct a RuntimeState, which is unacceptable.
So I think only the must-have things should be passed to it.

Reply:
+1

Reply:
resolved